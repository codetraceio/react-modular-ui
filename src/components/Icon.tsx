import * as React from 'react';

import {
  default as AbstractComponent,
  IModifiers
} from './AbstractComponent';
import settings from '../services/settingService';

export interface IIconProps {
  size?: string | number;
  width?: string | number;
  height?: string | number;
  color?: string;
  rotate?: string | number;
  name?: string;
  onClick?: (event: React.MouseEvent<HTMLDivElement>) => void;
}

export default class Button extends AbstractComponent<IIconProps, {}> {
  getModifierObject(): IModifiers {
    return {
      size: this.props.size,
      width: this.props.width,
      height: this.props.height,
      color: this.props.color,
      rotate: this.props.rotate,
      clickable: !!this.props.onClick,
    };
  }

  render() {
    return (
      <div
        className={this.getBlockName('icon', this.getModifierObject())}
        onClick={this.props.onClick}
      >
        {settings.getIcon(this.props.name)}
      </div>
    )
  }
}
