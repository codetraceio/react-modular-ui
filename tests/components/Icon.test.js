import * as React from "react";
import * as renderer from "react-test-renderer";

import Icon from "../../src/components/Icon";

describe("Icon", () => {
  it("should render Icon", () => {
    const tree = renderer.create(
      <Icon
        size="32"
        width="32"
        height="32"
        color="primary"
        rotate="180"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
