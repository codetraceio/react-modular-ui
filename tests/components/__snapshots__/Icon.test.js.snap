// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render Icon 1`] = `
<div
  className="c-icon -size-32 -width-32 -height-32 -color-primary -rotate-180"
/>
`;
